### PR TITLE
Allow configuration file to be specified when running statestore-checker

### PR DESCRIPTION
**Pull request by kadaan**
I store the state in S3, so I need to be able to pass authentication parameters to the statestore-checker.  To do this, I have added the ability to pass a config file to the statestore-checker whose properties are used to instantiate the hadoop configuration.

### CODE REVIEW

**Comment by chavdar** [comment]
@zliu41 @ibuenros  could you review please?

**Comment by ibuenros** [comment]
LGTM

File: gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java [hunk 1]
```diff
@@ -238,4 +265,11 @@ public static void main(String[] args)
       System.out.println(stringWriter.toString());
     }
   }
+
+  private static Properties fileToProperties(String fileName) throws IOException, ConfigurationException {
```
Comment (sahilTakiar) [inline gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java]: Can you move this method to `JobConfigurationUtils.java` it seems to be the same as `CliOptions.fileToProperties`.

File: gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java [hunk 2]
```diff
@@ -55,9 +63,16 @@
...
   private final boolean keepConfig;
 
-  public JobStateToJsonConverter(String storeUrl, boolean keepConfig)
+  public JobStateToJsonConverter(Properties props, String storeUrl, boolean keepConfig)
       throws IOException {
-    this.jobStateStore = new FsDatasetStateStore(storeUrl);
+    Configuration conf = new Configuration();
+    for (String key : props.stringPropertyNames()) {
+      conf.set(key, props.getProperty(key));
+    }
```
Comment (sahilTakiar) [inline gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java]: You can use `JobConfigurationUtils.putPropertiesIntoConfiguration` for this.

**Comment by sahilTakiar** [comment]
Minor comments, generally LGTM.

**Comment by sahilTakiar** [comment]
LGTM. Merging.

**Pull request merged**